Foundation/NSEnergyFormatter.swift

```diff
@@ -7,41 +7,195 @@
 // See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
 //
 
-
 extension EnergyFormatter {
-    public enum Unit : Int {
+    public enum Unit: Int {
+
+        case joule = 11
+        case kilojoule = 14
+        case calorie = 1793 // chemistry "calories", abbr "cal"
+        case kilocalorie = 1794 // kilocalories in general, abbr “kcal”, or “C” in some locales (e.g. US) when usesFoodEnergy is set to YES
+
+        // Map Unit to UnitEnergy class to aid with conversions
+        fileprivate var unitEnergy: UnitEnergy {
+            switch self {
+            case .joule:
+                return UnitEnergy.joules
+            case .kilojoule:
+                return UnitEnergy.kilojoules
+            case .calorie:
+                return UnitEnergy.calories
+            case .kilocalorie:
+                return UnitEnergy.kilocalories
+            }
+        }
+
+        // Reuse symbols defined in UnitEnergy, except for kilocalories, which is defined as "kCal"
+        fileprivate var symbol: String {
+            switch self {
+            case .kilocalorie:
+                return "kcal"
+            default:
+                return unitEnergy.symbol
+            }
+        }
+
+        // Return singular, full string representation of the energy unit
+        fileprivate var singularString: String {
+            switch self {
+            case .joule:
+                return "joule"
```

Comment:
We still need to identify a roadmap forward to handle localization bundles and the content of those for swift-corelibs-foundation. Since the environment is different (there is no "system" version of Foundation) we have to consider a methodology to store the data without relying on a common path (the bundle of the Foundation.framework). So I would say that the localization data is probably more an additional task than just the formatters. Thankfully code is mutable and we can change it after additions like this to accommodate for improvements in localizations.